extensions/rtmp/src/main/java/io/github/thibaultbee/streampack/ext/rtmp/streamers/CameraDualLiveStreamer.kt

```diff
@@ -0,0 +1,406 @@
+import io.github.thibaultbee.streampack.data.Config
+import io.github.thibaultbee.streampack.data.VideoConfig
+import io.github.thibaultbee.streampack.error.StreamPackError
+import io.github.thibaultbee.streampack.ext.rtmp.internal.endpoints.RtmpProducer
+import io.github.thibaultbee.streampack.internal.data.Frame
+import io.github.thibaultbee.streampack.internal.data.Packet
+import io.github.thibaultbee.streampack.internal.encoders.AudioMediaCodecEncoder
+import io.github.thibaultbee.streampack.internal.encoders.IEncoderListener
+import io.github.thibaultbee.streampack.internal.encoders.MultiVideoMediaCodecEncoder
+import io.github.thibaultbee.streampack.internal.endpoints.FileWriter
+import io.github.thibaultbee.streampack.internal.events.EventHandler
+import io.github.thibaultbee.streampack.internal.muxers.IMuxer
+import io.github.thibaultbee.streampack.internal.muxers.IMuxerListener
+import io.github.thibaultbee.streampack.internal.muxers.flv.FlvMuxer
+import io.github.thibaultbee.streampack.internal.muxers.flv.tags.video.ExtendedVideoTag
+import io.github.thibaultbee.streampack.internal.muxers.mp4.MP4Muxer
+import io.github.thibaultbee.streampack.internal.orientation.ISourceOrientationProvider
+import io.github.thibaultbee.streampack.internal.sources.AudioSource
+import io.github.thibaultbee.streampack.internal.sources.camera.CameraSource
+import io.github.thibaultbee.streampack.listeners.OnConnectionListener
+import io.github.thibaultbee.streampack.listeners.OnErrorListener
+import io.github.thibaultbee.streampack.logger.Logger
+import io.github.thibaultbee.streampack.streamers.settings.BaseCameraStreamerSettings
+import io.github.thibaultbee.streampack.streamers.settings.BaseStreamerAudioSettings
+import kotlinx.coroutines.runBlocking
+import java.io.File
+import java.nio.ByteBuffer
+import java.util.UUID
+
+
+class CameraDualLiveStreamer(
+    private val context: Context,
+    initialOnErrorListener: OnErrorListener? = null,
+    initialOnConnectionListener: OnConnectionListener? = null
+) : EventHandler() {
+
+    private enum class EncoderIndex(val index: Int) {
+        RTMP(0),
+        FILE(1)
+    }
+
+    private val rtmpProducer =
+        RtmpProducer().apply { onConnectionListener = initialOnConnectionListener }
+    private val fileWriter = FileWriter()
+    private var onErrorListener: OnErrorListener? = initialOnErrorListener
+
+    private val audioSource = AudioSource()
+    private val cameraSource = CameraSource(context)
+    private val rtmpMuxer: IMuxer = FlvMuxer(writeToFile = false)
+    private val fileMuxer: IMuxer = MP4Muxer()
+
+    private var rtmpAudioStreamId: Int? = null
+    private var rtmpVideoStreamId: Int? = null
+    private var fileAudioStreamId: Int? = null
+    private var fileVideoStreamId: Int? = null
+
+    // Keep video configuration separate for rtmp and file
+    private var rtmpVideoConfig: VideoConfig? = null
+    private var fileVideoConfig: VideoConfig? = null
```

Comment:
All these things, instead of being two of everything, should get grouped into a thing that there are two of. 

If done right, would lead eventually to a CameraMultiStreamer that could use any N streaming destinations.